Feature/sklearnex and remove faiss (#33)
Imageomics/emb-explorer: docs/BACKEND_PIPELINE.md

Author: ```suggestion
nice-to-have, never a hard requirement.

[^1]: [`sklearn-intelex`](https://github.com/uxlfoundation/scikit-learn-intelex) is powered by the [oneDAL](https://github.com/uxlfoundation/oneDAL) library that provides accelerations on x86_64 Linux and Windows machines, and silently fall back to vanilla `sklearn` on unsupported architectures like Apple Silicon and ARM Linux. The package is under the [UXL Foundation](https://github.com/uxlfoundation) (a Linux Foundation project) so cross-vendor support is a stated goal.
```

Reviewer: ```suggestion

Note that `sklearn-intelex` acceleration is used for CPU operations where available[^1].

```

  ├─ L2 Normalize: project onto unit hypersphere
  │
  ├─► Step 1: KMeans Clustering (high-dimensional)

>   │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by sklearn-intelex)

Author: ```suggestion
  │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by `sklearn-intelex` [^1])
```

  │
  ├─► Step 2: Dimensionality Reduction to 2D
  │     Method:  PCA / t-SNE / UMAP

>   │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by sklearn-intelex for PCA/TSNE)

Author: ```suggestion
  │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by `sklearn-intelex` for PCA/TSNE [^1])
```

>   │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by `sklearn-intelex` for PCA/TSNE [^1])

Reviewer: ```suggestion
  │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by `sklearn-intelex`)
 │
 ├─► Step 2: Dimensionality Reduction to 2D
 │     Method:  PCA / t-SNE / UMAP
  │     Backend: cuML (GPU) → sklearn (CPU, auto-accelerated by `sklearn-intelex` for PCA/TSNE)
```
Wrong spot for footnotes (they won't function in the codeblock)

| Backend | When It's Used | How It Works |
|---------|---------------|--------------|
| **cuML** | GPU available + >500 samples | GPU-accelerated KMeans via RAPIDS. Runs on CuPy arrays. Falls back to sklearn on any error. |

> | **sklearn** | CPU path (default on machines without a GPU) | Standard scikit-learn KMeans, auto-accelerated by [scikit-learn-intelex](https://github.com/uxlfoundation/scikit-learn-intelex) (Intel oneDAL) when installed — typically 10–17× faster than vanilla sklearn on CPU. Disable with `EMB_EXPLORER_DISABLE_SKLEARNEX=1`. |

Reviewer: ```suggestion
| **sklearn** | CPU path (default on machines without a GPU) | Standard scikit-learn KMeans, auto-accelerated by [scikit-learn-intelex](https://github.com/uxlfoundation/scikit-learn-intelex) (Intel oneDAL) when installed[^1] — typically 10–17× faster than vanilla sklearn on CPU. Disable with `EMB_EXPLORER_DISABLE_SKLEARNEX=1`. |
```

| Operation | Auto Logic |
|-----------|-----------|

> | KMeans | cuML if GPU + >500 samples, else sklearn (auto-accelerated by sklearn-intelex when installed) |

Author: ```suggestion
| KMeans | cuML if GPU + >500 samples, else sklearn (auto-accelerated by `sklearn-intelex` when installed [^1]) |
```

> | KMeans | cuML if GPU + >500 samples, else sklearn (auto-accelerated by `sklearn-intelex` when installed [^1]) |

Reviewer: ```suggestion
| KMeans | cuML if GPU + >500 samples, else sklearn (auto-accelerated by `sklearn-intelex` when installed[^1]) |
```
